app/controllers/gigs_controller.rb

```diff
@@ -0,0 +1,65 @@
+class GigsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :require_admin, only: [ :manage, :new, :create, :edit, :destroy ]
+  before_action :set_gig,       only: [ :show, :edit, :update ]
+
+  def new
+    @gig = Gig.new
+  end
+
+  def create
+    @gig = Gig.new(allowed_gig_params)
+    if @gig.save
```

Comment:
This is not wrong, but I've found myself enjoying using the `responders` gem, that automates/boilerplates most of this sort of logic.